diff -Naur original\sox/libgsm/CMakeLists.txt packages\sox/libgsm/CMakeLists.txt
--- original\sox/libgsm/CMakeLists.txt	Wed Mar  2 22:10:27 2011
+++ packages\sox/libgsm/CMakeLists.txt	Thu Aug 20 10:59:03 2015
@@ -1 +1,9 @@
 add_library(gsm add code decode gsm_create gsm_decode gsm_destroy gsm_encode gsm_option long_term lpc preprocess rpe short_term table)
+
+# Set the Output Directory for libs
+set_target_properties( gsm
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+)
diff -Naur original\sox/lpc10/CMakeLists.txt packages\sox/lpc10/CMakeLists.txt
--- original\sox/lpc10/CMakeLists.txt	Sat Sep 24 19:02:25 2011
+++ packages\sox/lpc10/CMakeLists.txt	Thu Aug 20 10:56:10 2015
@@ -2,3 +2,11 @@
 
 add_library(lpc10 analys bsynz chanwr dcbias decode deemp difmag dyptrk encode energy f2clib ham84 hp100 invert irc2pc ivfilt lpcdec lpcenc lpcini lpfilt median mload onset pitsyn placea placev preemp prepro random rcchk synths tbdm voicin vparms)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/../src)
+
+# Set the Output Directory for libs
+set_target_properties( lpc10
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+)
diff -Naur original\sox/src/CMakeLists.txt packages\sox/src/CMakeLists.txt
--- original\sox/src/CMakeLists.txt	Thu Oct 30 03:29:54 2014
+++ packages\sox/src/CMakeLists.txt	Thu Aug 20 10:54:08 2015
@@ -2,6 +2,9 @@
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
+# Add the LibOutput directory into the linker path for picking up other libs
+link_directories(${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput)
+
 if(CMAKE_COMPILER_IS_GNUCC)
   execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion
     OUTPUT_VARIABLE ver)
@@ -76,6 +79,7 @@
   vol
 )
 set(formats_srcs
+  swig_wrap.c
   8svx
   adpcm
   adpcms
@@ -140,13 +144,25 @@
 #set(formats_srcs ${formats_srcs} coreaudio sndio sunaudio waveaudio)
 #add_definitions(-Ibit-rot)
 
-add_library(lib${PROJECT_NAME}
+add_library(lib${PROJECT_NAME} SHARED
   effects                 formats_i               libsox_i
   effects_i               ${formats_srcs}         ${optional_srcs}
   effects_i_dsp           getopt
   ${effects_srcs}         util
   formats                 libsox                  xmalloc
 )
+
+# Add the depend libs into the linker
+target_link_libraries(lib${PROJECT_NAME} gsm.lib lpc10.lib)
+
+# Set the Output Directory for libs
+set_target_properties( lib${PROJECT_NAME}
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../LibOutput"
+)
+
 add_executable(${PROJECT_NAME} ${PROJECT_NAME}.c)
 target_link_libraries(${PROJECT_NAME} lib${PROJECT_NAME} lpc10 ${optional_libs})
 add_executable(sox_sample_test sox_sample_test.c)
